src/coreComponents/physicsSolvers/NonlinearSolverParameters.cpp

```diff
@@ -74,6 +74,11 @@ NonlinearSolverParameters::NonlinearSolverParameters( string const & name,
     setApplyDefaultValue( 0 ).
     setDescription( "Number of Newton's iterations." );
 
+  registerWrapper( viewKeysStruct::maxAllowedResidualNormString, &m_maxAllowedResidualNorm ).
+    setApplyDefaultValue( 1e9 ).
```

Comment:
In fact we don't want something as large as the max value of `real64` because we want the solver to cut the time step before the residual norm reaches these very large values. 

Here, I chose `1e9` because some solvers (mostly the `SurfaceGenerator`) commonly have a very large residual norm, but for typical field-scale reservoir simulations `1e9` is already too large, so I would probably use the `maxAllowedResidualNorm` to reduce this value by hand.